src/esp/sim/RenderInstanceHelper.h

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) Meta Platforms, Inc. and its affiliates.
+// This source code is licensed under the MIT license found in the
+// LICENSE file in the root directory of this source tree.
+
+#ifndef ESP_SIM_RENDERINSTANCEHELPER_H_
+#define ESP_SIM_RENDERINSTANCEHELPER_H_
+
+#include <string>
+#include <vector>
+
+#include "esp/core/Esp.h"
+
+namespace esp {
+namespace scene {
+class SceneNode;
+}
+namespace sim {
+class Simulator;
+}
+}  // namespace esp
+
+namespace esp {
+namespace sim {
+
+class RenderInstanceHelper {
+ public:
+  RenderInstanceHelper(Simulator& sim, bool useXYZWOrientations = true);
+
+  int AddInstance(const std::string& assetFilepath, int semanticId);
+  void ClearAllInstances();
+  int GetNumInstances();
+  void SetWorldPoses(float* positions,
+                     size_t positionsSize,
+                     float* orientations,
+                     size_t orientationsSize);
+
+ private:
+  Simulator* sim_ = nullptr;
+  std::vector<scene::SceneNode*> instances_;
```

Comment:
We should consider creating a `GetInstanceNode(int index)` function to enable handling bounding boxes and the like. This can be done later as needed.

Reply:
On the contrary, I don't intend to expose Habitat SceneNodes! One design goal here is to make it easy to port RenderInstanceHelper to a different renderer backend, e.g. the Magnum batch renderer or the Isaac Sim renderer.